publication-rest/src/main/java/no/unit/nva/publication/download/CreatePresignedDownloadUrlHandler.java

```diff
@@ -80,47 +80,38 @@ protected void validateRequest(Void unused, RequestInfo requestInfo, Context con
-        var file =  publication.getFile(fileIdentifier);
+        var publication = resourceService.getResourceByIdentifier(publicationId);
+        var file = publication.getFileEntry(fileIdentifier);
 
         if (file.isEmpty() || !hasFileAccess(publication, file.get(), requestInfo)) {
             throw new NotFoundException(
                 String.format(REQUESTED_RESOURCE_NOT_FOUND, publication.getIdentifier(), fileIdentifier));
         }
 
-        return getPresignedUriResponse(file.get());
+        return getPresignedUriResponse(file.get().getFile());
     }
 
     private PresignedUriResponse getPresignedUriResponse(File file) throws S3ServiceException {
         var expiration = defaultExpiration();
         var preSignedUriLong = getPresignedDownloadUrl(file, expiration);
         var shortenedPresignUri = uriShortener.shorten(preSignedUriLong.id(), basePath, expiration);
-        return new PresignedUriResponse(file.getIdentifier().toString(), preSignedUriLong.id(), expiration, shortenedPresignUri);
+        return new PresignedUriResponse(file.getIdentifier().toString(), preSignedUriLong.id(), expiration,
+                                        shortenedPresignUri);
     }
 
-    private boolean hasFileAccess(Publication publication, File file, RequestInfo requestInfo) {
-        return userHasUpdatePermission(publication, requestInfo) || fileIsVisibleForNonOwner(publication, file);
+    private boolean hasFileAccess(Resource publication, FileEntry file, RequestInfo requestInfo) {
+        var userInstance = getUserInstance(requestInfo);
+        var permissions = FilePermissions.create(file, userInstance, publication);
+        return permissions.allowsAction(FileOperation.DOWNLOAD);
     }
 
-    private boolean fileIsVisibleForNonOwner(Publication publication, File file) {
-        return isPublicationPublished(publication) && file.isVisibleForNonOwner();
-    }
-
-    private boolean isPublicationPublished(Publication publication) {
-        return PublicationStatus.PUBLISHED.equals(publication.getStatus());
-    }
-
-    private boolean userHasUpdatePermission(Publication publication, RequestInfo requestInfo) {
-        return attempt(() -> RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient)).toOptional()
-                   .map(user -> hasUpdatePermission(publication, user))
-                   .orElse(false);
-    }
-
-    private boolean hasUpdatePermission(Publication publication, UserInstance user) {
-        return PublicationPermissions.create(publication, user)
-                   .allowsAction(PublicationOperation.UPDATE);
+    private UserInstance getUserInstance(RequestInfo requestInfo) {
+        return attempt(() ->
+                           RequestUtil.createUserInstanceFromRequest(requestInfo,
+                                                                     identityServiceClient))
+                   .toOptional().orElse(null);
```

Comment:
Not sure about `toOptional().orElse(null)` What is the effect of `toOptional` here? Do we later throw? If so, throw here. Can the UserInstance simply be returned?

Reply:
I want to convert UnauthorizedException into a null value, FilePermissions.create can be used by anonymous users